iceoryx_binding_c/include/iceoryx_binding_c/publisher.h

```diff
@@ -24,20 +24,29 @@
 /// @brief publisher handle
 typedef struct cpp2c_Publisher* iox_pub_t;
 
+/// @brief options to be set for a publisher
+typedef struct
+{
+    // size of the history chunk queue
+    uint64_t historyCapacity;
+
+    // name of the node the publisher belongs to
+    const char* nodeName;
```

Comment:
how about using a `char nodeName[100]`?

On the other hand one cannot assign `const char*` like `nodeName = "foo";` anymore.

Reply:
Yes but you can still use a one liner like `strncpy(nodeName, "foo",  100);`.
You are right this is not the most beautiful way but we are working here also in a safety context ...

Reply:
yes, but it's more error prone like `strncpy(myNodeName, options.nodeName, 100);` and then needing to debug the code because of this mistake that would not be possible if it would remain a `const char*`

Reply:
Rejected. Usability suffers for not much (if any) gain.